src/compiler/scala/tools/nsc/backend/jvm/PostProcessor.scala

```diff
@@ -75,17 +76,24 @@ abstract class PostProcessor extends PerRunInit {
       classfileWriter.write(internalName, bytes, paths)
     }
   }
+
   private def warnCaseInsensitiveOverwrite(clazz: GeneratedClass): Unit = {
     val name = clazz.classNode.name
     val lowercaseJavaClassName = name.toLowerCase
-    val sourceClassName = clazz.sourceClassName
-
-    val duplicate = caseInsensitively.putIfAbsent(lowercaseJavaClassName, sourceClassName)
-    if (duplicate != null) {
-      backendReporting.warning(
-        clazz.position,
-          s"Class ${sourceClassName} differs only in case from ${duplicate}. " +
-            "Such classes will overwrite one another on case-insensitive filesystems."
+
+    caseInsensitively.get(lowercaseJavaClassName) match {
+      case null =>
+        caseInsensitively.put(lowercaseJavaClassName, (name, clazz.position))
```

Comment:
this code can can be executed concurrently, so we should use an atomic operation

Reply:
Done.